ddspipe_participants/src/cpp/writer/rtps/CommonWriter.cpp

```diff
@@ -125,6 +125,16 @@ void CommonWriter::onWriterMatched(
     }
 }
 
+void CommonWriter::onWriterChangeReceivedByAll(
+        fastrtps::rtps::RTPSWriter* /*writer*/,
+        fastrtps::rtps::CacheChange_t* change)
+{
+    if (writer_qos_.m_durability.kind == fastdds::dds::VOLATILE_DURABILITY_QOS)
```

Comment:
Or is best effort. The combination of best_effort + transient makes no sense but may occur.